OpenSim/Simulation/Model/BushingForce.cpp

```diff
@@ -47,27 +47,95 @@ BushingForce::BushingForce()
     constructProperties();
 }
 
-// Convenience constructor.
-BushingForce::BushingForce(const string&    body1Name, 
-                           const Vec3&      point1, 
-                           const Vec3&      orientation1,
-                           const string&    body2Name, 
-                           const Vec3&      point2, 
-                           const Vec3&      orientation2,
-                           const Vec3&      transStiffness, 
-                           const Vec3&      rotStiffness, 
-                           const Vec3&      transDamping, 
-                           const Vec3&      rotDamping)
+/* Construct a BusingForce by supplying the Offset frames that the BushingForce
+   acts between */
+BushingForce::BushingForce(const PhysicalOffsetFrame& frame1,
+    const PhysicalOffsetFrame& frame2,
+    const SimTK::Vec3& transStiffness,
+    const SimTK::Vec3& rotStiffness,
+    const SimTK::Vec3& transDamping,
+    const SimTK::Vec3& rotDamping)
+{
+    set_offset_frame1(frame1);
+    set_offset_frame2(frame2);
+    set_rotational_stiffness(rotStiffness);
+    set_translational_stiffness(transStiffness);
+    set_rotational_damping(rotDamping);
+    set_translational_damping(transDamping);
+}
+
+/* Convenience construction that creates the Offsets on the Physical Frames (e.g.
+   Bodies that the BushingForce acts between. */
+BushingForce::BushingForce(const PhysicalFrame& frame1,
+    const SimTK::Vec3& point1,
+    const SimTK::Vec3& orientation1,
+    const PhysicalFrame& frame2,
+    const SimTK::Vec3& point2,
+    const SimTK::Vec3& orientation2,
+    const SimTK::Vec3& transStiffness,
+    const SimTK::Vec3& rotStiffness,
+    const SimTK::Vec3& transDamping,
+    const SimTK::Vec3& rotDamping)
 {
     setNull();
     constructProperties();
 
-    set_body_1(body1Name);
-    set_body_2(body2Name);
-    set_location_body_1(point1);
-    set_orientation_body_1(orientation1);
-    set_location_body_2(point2);
-    set_orientation_body_2(orientation2);
+    upd_offset_frame1().updConnector<PhysicalFrame>("parent")
+        .set_connected_to_name(frame1.getName());
+    Rotation rotation1(BodyRotationSequence,
+        orientation1[0], XAxis,
+        orientation1[1], YAxis,
+        orientation1[2], ZAxis);
+    upd_offset_frame1().setOffsetTransform(Transform(rotation1, point1));
+
+    upd_offset_frame2().updConnector<PhysicalFrame>("parent")
+        .set_connected_to_name(frame2.getName());
+    Rotation rotation2(BodyRotationSequence,
+        orientation2[0], XAxis,
+        orientation2[1], YAxis,
+        orientation2[2], ZAxis);
+    upd_offset_frame2().setOffsetTransform(Transform(rotation2, point2));
+
+    set_rotational_stiffness(rotStiffness);
```

Comment:
Lots of code duplication here between constructors. Better just to have one constructor I think, but if there has to be more than one the duplicate code should be factored out into an init() method or even better (since C++11) you can invoke one constructor from the other ones.